File: Codeforces/Rounds/Codeforces Round 970 (Div. 3)/B.  Square or Not.py
import math

def check_square(n):
    root = int(math.sqrt(n))
    return root * root == n, root

t = int(input())
for _ in range(t):
    l = int(input())
    s = input()
    check, root = check_square(l)
    if not check:
        print("NO")
    else:
        borders = 4 * (root - 1)
        zeros = l - borders
        zeros_count = s.count("0")
        if borders == l or zeros_count == zeros:
            print("YES")
        else:
            print("NO")

File: Codeforces/Rounds/Codeforces Round 970 (Div. 3)/B. Square or Not.cpp
# include <iostream>
#include <algorithm>
#include <cmath>
#include <string>
using namespace std;

int main(){
    int t; cin>>t;
    while(t--){
        int l, root, borders, zeros, zeros_count;
        string s;
        cin >> l >> s;
        root = sqrt(l);
        // check if square
        if (root*root != l){
            cout << "NO"<< endl;
        }
        else{
            borders = 4*(root - 1);
            zeros = l-borders;
            zeros_count = count(s.begin(), s.end(), '0');
            if (borders == l || zeros_count==zeros) cout << "YES"<< endl;
            else cout << "NO"<< endl;
        }
    }
    return 0;
}
